pkg/costmodel/costmodel.go

```diff
@@ -1181,6 +1181,14 @@ func (cm *CostModel) GetNodeCost(cp costAnalyzerCloud.Provider) (map[string]*cos
 					log.Warnf("Could not parse total node price")
 					return nil, err
 				}
+				if newCnode.GPUCost != "" {
+					gpuPrice, err := strconv.ParseFloat(newCnode.GPUCost, 64)
+					if err != nil {
+						log.Warnf("Could not parse node gpu price")
+						return nil, err
+					}
+					nodePrice = nodePrice - gpuPrice // remove the gpuPrice from the total, we're just costing out RAM and CPU.
```

Comment:
Based on the title of the PR ("GPU cost is defined, but not CPU/RAM) I'm trying to understand the contextual conditional scope of this change.

On line 1045 we start with with:
```go
if newCnode.GPU != "" && newCnode.GPUCost == ""
```

Then on 1135 we have:
```go
else if newCnode.RAMCost == ""
```

We're in that block. So we're really in the conditional scope:
```go
if !(newCnode.GPU != "" && newCnode.GPUCost == "") && newCnode.RAMCost == ""
```

Then within _that_ scope we hit:
```go
if newCnode.Cost != ""
```

So in total we're making this change when:
```go
if !(newCnode.GPU != "" && newCnode.GPUCost == "") && newCnode.RAMCost == "" && newCnode.Cost != ""
```
or equivalently
```go
if (newCnode.GPU == "" || newCnode.GPUCost != "") && newCnode.RAMCost == "" && newCnode.Cost != ""

In plain English, the relevant change would seem to happen when "(GPU **isn't** set OR GPUCost **is** set) and RAMCost **isn't** set, but Cost **is** set". So there's a cost, but no RAMCost, and either there is a GPUCost or there's not a GPU. Something like that.

Do I have that right? If so, and that makes sense, then I don't see any mistakes in the code. But the idea is still mostly opaque to me.

Anyways, I can approve this, but I'd be totally leaning on your testing. This is difficult to review.

Reply:
Yeah. Essentially this is saying-- I need to pull GPU costs in from a separate line item GPU and fulfill the given node price with them.

Reply:
The key here is that when total cost is set and RAM cost is not, you need to perform inference. WHen you have GPU you need to take that into account when doing your inference.